### PR TITLE
Partially rework of the entry form

### PR DESCRIPTION
**Pull request by auge8472**
This PR introduces several changes in the fieldset `#message` of the posting form.

The button bar moves from the right side to top of the textarea and the buttons are displayed in horizontal rules, one for the formatting buttons and one for the smilies. Behind the scenes also the description lists with the instructions for BB-codes and smilies for the use without active JavaScript moves to above the textarea. The list items (`dt`, `dd`) got enclosed in a dedicated div, what made it necessary to adapt this change to the posting.js. For immediately jump to the textarea, the instruction block got a skip link (currently hardcoded in german language).

Additionally the list of additional buttons was reworked from a collection of links to a list of buttons, what made it necessary to change also the event code for inserting the smiley code into the textarea.

### CODE REVIEW

**Comment by auge8472** [comment]
@loesler I would like to ask you specifically for a review of the changes in the Javascript source code. There are several styling fixes. The relevant functional changes was made in the functions `BBCodeSingleSmilieButton` and `ButtonGroup` as well as the anonyme functions for the variables `convertInstructionsToButton`, `convertInstructionsToSmilies`, `initTextArea` and `removeIntroductionElements`.

**Review by loesler — APPROVED** [review]
The JS looks fine and works for me

**Comment by auge8472** [comment]
@loesler Thank you for your review.

File: themes/default/subtemplates/posting.inc.tpl [hunk 1]
```diff
@@ -82,61 +82,61 @@
...
-</p>
+<input id="repeat_email" type="email" size="40" name="{$fld_repeat_email}" value="{if $honey_pot_email}{$honey_pot_email}{/if}" maxlength="{$settings.email_maxlength}" tabindex="-1" />
+</div>
 
-<p>
+<div>
 <label for="hp" class="input">{#hp_marking#}</label>
-<input id="hp" type="text" size="40" name="{$fld_hp}" value="{if $hp}{$hp}{/if}" maxlength="{$settings.hp_maxlength}" tabindex="3" />&nbsp;<span class="xsmall">{#optional#}</span>
-</p>
+<input id="hp" type="url" size="40" name="{$fld_hp}" value="{if $hp}{$hp}{/if}" maxlength="{$settings.hp_maxlength}" tabindex="3" />&nbsp;<span class="xsmall">{#optional#}</span>
```
Comment (loesler) [inline themes/default/subtemplates/posting.inc.tpl:102]: inside the label?

File: themes/default/subtemplates/posting.inc.tpl [hunk 2]
```diff
@@ -82,61 +82,61 @@
...
 <input id="name" type="text" size="40" name="{$fld_user_name}" value="{if $name}{$name}{/if}" maxlength="{$settings.username_maxlength}"  tabindex="1" />
-</p>
+</div>
 
-<p>
+<div>
 <label for="email" class="input">{#email_marking#}</label>
-<input id="email" type="text" size="40" name="{$fld_user_email}" value="{if $email}{$email}{/if}" maxlength="{$settings.email_maxlength}" tabindex="2" />&nbsp;<span class="xsmall">{#optional_email#}</span>
-</p>
+<input id="email" type="email" size="40" name="{$fld_user_email}" value="{if $email}{$email}{/if}" maxlength="{$settings.email_maxlength}" tabindex="2" />&nbsp;<span class="xsmall">{#optional_email#}</span>
```
Comment (loesler) [inline themes/default/subtemplates/posting.inc.tpl:92]: inside the label?

**Review by loesler — COMMENTED** [review]
Shouldn't the xsmall text be inside the label?

File: themes/default/subtemplates/posting.inc.tpl (hunk 1 above)
Comment (auge8472) [reply themes/default/subtemplates/posting.inc.tpl:102]: Oha!

Comment (loesler) [reply themes/default/subtemplates/posting.inc.tpl:102]: Let's take a look to the "tags" input field, the code is (currently)

```
<div>
<label for="tags" class="input">{#tags_marking#} <span class="xsmall">{#tags_note#}</span></label>
<input id="tags" type="text" size="50" name="tags" value="{$tags|default:""}" maxlength="253" tabindex="-1" />
</div>
```

The first _xsmall span_ is inside the label tag. If we compare this code to the e.g. location input, i.e.,

```
<div>
<label for="location" class="input">{#location_marking#}</label>
<input id="location" type="text" size="40" name="{$fld_location}" value="{if $location}{$location}{/if}" maxlength="{$settings.location_maxlength}" tabindex="4" />&nbsp;<span class="xsmall">{#optional#}</span>
</div>
```

you will notice that the _xsmall span_ is not inside the label tag, but behind the input.   I wonder why the two pieces of code are different.

Comment (auge8472) [reply themes/default/subtemplates/posting.inc.tpl:102]: Ok, now I understand your concern. I will correct the code.

**Comment by auge8472** [comment]
The help notices (I found three items) moved to the label, like it's done for the tags input before. Additionally – not shown on the screenshot in this comment – the checkbox for managing a cookie to store the userdata in the browser[^1] – moved to the start of the line.

This is a screenshot of a part of the posting form. The help notices (like "optional") are now part of the input label

[^1]: I think, this has to be redone. Storing the userdata of unregistered visitors in a cookie, that is sent from the browser to the server with every request, is a big privacy issue. We can store it in the browser (`localStorage`) but this makes it working only in the specific browser on the specific device. If one uses the forum with different browsers on different devices, one has to save the data multiple times. Alternatively we can remove this featurette. This would be an incentive to sign up in a forum.

**Comment by auge8472** [comment]
@loesler: I'm sorry for poking at you again. Can you please provide the YUM-compressed versions for `js/main.js` and `js/posting.js` in a separate PR? Thank you in advance.

**Comment by loesler** [comment]
> Alternatively we can remove this featurette. 

That is my favorite. In my forum, this option is disabled, i.e. `remember_last_visit = 0` and `remember_userdata = 0` 

>  Can you please provide the YUM-compressed versions for js/main.js and js/posting.js in a separate PR? Thank you in advance.
I will try to compress the Script, yes.

**Comment by auge8472** [comment]
> > Alternatively we can remove this featurette.
> 
> That is my favorite.

Full ACK.

**Pull request merged**